### PR TITLE
Support ansible variable in playbook partially

### PR DESCRIPTION
**Pull request by okamototk**
Support ansible variables to avoid duplicate parameter definitions over ansible and ansible_spec.

### CODE REVIEW

**Comment by volanja** [comment]
Thank you send PR.Your ideas looks good!
But, `lib/src/*` is difficult to test..
So, As much as possible, I do not want to change the `lib/src/*`
I want to change lib/* (without lib/src/*) instead of these files. Because of testability.
Do you have some other ideas?

**Comment by okamototk** [comment]
I moved most code to lib/*. 
However, code loads configuration files and I don't have any idea to write test now.
Please check updated commit.

**Comment by okamototk** [comment]
I published sample playbook here: https://github.com/okamototk/ansible_spec_variable_sample/.

**Comment by volanja** [comment]
Thank you your contribution.
I checked code, I am writing some test case before merge. 
I refer to your sample playbook.

File: lib/ansible_spec/load_ansible.rb [hunk 1]
```diff
@@ -287,4 +288,60 @@ def self.get_properties()
...
+    vars_file = 'group_vars/all.yml'
+    if File.exist?(vars_file)
+      yaml = YAML.load_file(vars_file)
+      if yaml.kind_of?(Hash)
+        vars.merge!(yaml)
+      end
+    end
+
+    # each group vars
+    if p[group_idx].has_key?('group')
```
Comment (volanja) [inline lib/ansible_spec/load_ansible.rb]: I think that this code can't work.
Because p[grorp_idx] don' t have 'group' key.

Comment (okamototk) [inline lib/ansible_spec/load_ansible.rb]: I missed group handling when refactor the code.
I fixed and add lacked role default variable handling.

thanks,

**Comment by okamototk** [comment]
Travis showed error because hash sized is changed to get group.
If you agree my proposal, I'll fix the tests.

**Comment by volanja** [comment]
Off course. Please fix test.
(Sorry, I made a mistake some comment in test..)

**Comment by volanja** [comment]
Thank you your contribution :)
I add test and released [v0.2.8](https://rubygems.org/gems/ansible_spec/versions/0.2.8)

**Comment by okamototk** [comment]
thanks!

I also send pull request for ansible-sample-tdd: https://github.com/volanja/ansible-sample-tdd/pull/4
Please check it.

**Comment by volanja** [comment]
I merged it now.

**Pull request merged**